overlord/snapstate/snapstate.go

```diff
@@ -820,7 +820,20 @@ func RestartSystem(task *state.Task, rebootInfo *boot.RebootInfo) {
 	if immediate {
 		rst = restart.RestartSystemNow
 	}
-	restart.Request(task.State(), rst, rebootInfo)
+
+	return FinishTaskWithRestart(task, status, rst, rebootInfo)
+}
+
+// FinishTaskWithRestart will finish any task that needs a restart.
+// TODO for the moment it just calls restart.Request, but we will
+// return an error in the future if we want to hold the restart. We
+// will also add post hold status and boot id information in the task
+// in that case.
```

Comment:
I suggest reformulating this by having a single TODO line like this

    TODO: the part below has not been implemented yet

and write the rest of the comment as if it was the documentation for the function.

Reply:
Done